app/controllers/posts_controller.rb

```diff
@@ -1,13 +1,13 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    @posts = User.find(params[:user_id]).posts
+    @posts = @user.posts
   end
 
   def show
     @user = User.find(params[:user_id])
     @post = Post.find(params[:id])
-    @comments = @post.comments
+    @comments = @post.comments.includes([:author])
   end
 
   def new; end
```

Comment:
- [ ] Kindly solve the  N+1 problem when fetching all posts and their comments for a user.

   - [HINT] In the `posts#index` actions, you can use:
   ```
   @posts = @user.posts.includes(:comments)
   ```
   instead of 
   ```
   @posts = @user.posts
   ```

Reply:
Thanks for the tip, It would definitely improve the querying performance. 💯